### PR TITLE
JavaScript/Router add knowledge checks and learning outcomes

### LINKED ISSUE #22801: Add missing Knowledge Checks, Learning Outcomes & answer-links to JavaScript/Router
Related to #22472

This task is to go through the lesson referenced in the title of this issue and (as applicable):

- Add missing Learning Outcomes
- Add missing Knowledge Checks
- Add answer links in the Knowledge Checks

If the lesson already has Learning Outcomes or Knowledge Checks they do not need to be added to. Many lessons only need answer links added.

Knowledge check links should consist of an `<a>` tag which adds the `knowledge-check-link` class and links to the part of the lesson (or the external resource that is linked in the lesson) where the answer to the Knowledge Check can be found. Knowledge Check links should **not** point to external resources that weren't linked to as part of the lesson content or assignment. You can link to a specific line in the lesson like so:

```html
Lorem ipsum dolor sit amet, <span id="js-function-knowledge-check">consectetur adipiscing elit</span>, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to the page where the span with this id lives -->
<a class="knowledge-check-link" href="#js-function-knowledge-check">How do you return a value from a function?</a>
```
Or you can link to just a section of the lesson by just linking to the section heading and following [these](https://github.com/TheOdinProject/curriculum/wiki/Contributing-to-Wiki#anchor-linking) rules for determining how to format the link. Example:

```html
### Lorem ipsum dolor sit amet

sed do eiusmod tempor incididunt ut labore et dolore magna aliqua

...
<!-- This a tag just goes to a heading on the page-->
<a class="knowledge-check-link" href="#lorem-ipsum-dolor-sit-amet">How do you return a value from a function?</a>
```

### Acceptance Criteria

- [ ] Lesson referenced in the title has Learning Outcomes, Knowledge Checks, and Knowledge Check links
- [ ] Knowledge Check links all link to the proper place in the lesson, or the proper external resource that was linked to earlier in the lesson
- [ ] Knowledge Check styling should **not** change

Please do not start work on this unless you've been assigned to it. Comment on this issue and ping me @wise-king-sullyman if you would like to be assigned to it.

Additionally, we would like for contributors to only take on one of these issues at a time. Please only request assignment for (and begin work on) an issue after the Learning Outcome/Knowledge Check/Knowledge Check link issue you're currently assigned to has been approved and merged.

Please check out [this new post on our wiki](https://github.com/TheOdinProject/curriculum/wiki/Learning-Outcomes-vs-Knowledge-Checks) about Learning Outcomes and Knowledge Checks if you need to write new Learning Outcomes or Knowledge Checks for this issue.

### PR DESCRIPTION
**Pull request by digidub**
- [x] You have read [The Odin Projects Contributing Guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md).
 - [x] The PR title summarizes the change and where it happened, for example: "Fixes punctuation in Clean Code lesson".
 
#### 1.Describe the changes made and include why they are necessary or important:

Added knowledge checks and learning outcomes for JS Router lesson

#### 2. If this PR is related to an open issue please reference it with the `#` operator and the issue number (and any [relevant keywords](https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests)) below:

closes #22801

### CODE REVIEW

File: javascript/react-js/router.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
 
 In this lecture we are going to learn a little bit about routing in React. Up until now we have only written one-page applications. Once your application has multiple pages, you need to set up a reliable routing system. This is to handle the component or page that should be rendered when navigating to a certain route. For this we are going to use a package called `react-router-dom`.
 
+### Learning Outcomes
+By the end of this lesson, you should be able to:
+
+- Explain how you add page routing to a React project
```
Comment (twalton83) [inline javascript/react-js/router.md:8]:
```suggestion
- Explain how to add page routing to a React project
```

File: javascript/react-js/router.md [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
+- Explain the core three components of React Router
```
Comment (twalton83) [inline javascript/react-js/router.md:9]:
```suggestion
- Explain the three core components of React Router
```

File: javascript/react-js/router.md [hunk 3]
```diff
@@ -111,3 +120,10 @@ This section contains helpful links to other content. It isn't required, so cons
 
 - [This video by The Net Ninja](https://www.youtube.com/watch?v=QUz3k2O3ZJU&ab_channel=TheNetNinja) is an excellent review of what we learned here.
 - For some extra practice/review, check out [this article](https://css-tricks.com/learning-react-router/).
+
+### Knowledge Checks
+
+- <a class="knowledge-check-link" href="#react-router">How do you add page routing to a React project?</a>
```
Comment (twalton83) [inline javascript/react-js/router.md:126]: We don't want to just wholesale copy the Learning Outcomes for our Knowledge Checks, they should be difference.

**Review by twalton83 — CHANGES_REQUESTED** [review]

File: javascript/react-js/router.md (hunk 3 above)
Comment (digidub) [reply javascript/react-js/router.md:126]: Thanks for the feedback, I appreciate the clarification. I'll make some changes and resubmit :+1:

**Comment by digidub** [comment]
@twalton83 I've updated the Learning Outcomes to fit your guidance with them covering "what a student should know, broadly". Since this lesson is fairly small in content scope and simply covers the `react-router-dom` package, I've just covered that off in the Learning Outcomes - since in my view that is the broad goal of the lesson. I've kept the Knowledge Checks unchanged, as they are fairly specific.

If you have any suggestions for anything else to add I'm very happy to consider.

Thanks for your help.

**Comment by twalton83** [comment]
We're pointing right to the same header, I think we should point our links to a more specific section in my opinion.

**Comment by digidub** [comment]
@twalton83 Thanks, I've added span tags for improved link accuracy on the knowledge checks. Let me know if you'd like to see any more changes! :)

**Comment by twalton83** [comment]
If you fix the merge conflicts we should be good to go.

**Review by twalton83 — APPROVED** [review]

**Comment by digidub** [comment]
@twalton83 that should all be done now :)

**Pull request merged**